asmcomp/cmm.ml

```diff
@@ -223,6 +223,31 @@ let ccatch (i, ids, e1, e2, dbg) =
 let reset () =
   label_counter := 99
 
+let lift_phantom_lets f exp =
+  let rec lift exp lifted_rev depth =
+    let next_depth = depth + 1 in
+    match exp with
+    | Cop (op, args, dbg) when depth < 4 ->
```

Comment:
What is the purpose of depth here? Maybe add a comment why we need it.

Reply:
In theory, unbounded exploration is linear in the size of the expression, which can lead to a translation pass that becomes quadratic in the size of the original program. Limiting to a fixed depth prevents this problem, and it happens that no existing optimisation looks deeper than 3 nested operations, so no optimisation is lost because of this restriction. I'll add a comment explaining this.